testsuite/classlibrary/TestPatternProxy.sc

```diff
@@ -362,8 +362,31 @@ TestPatternProxy : UnitTest {
 
 		Pdefn.clear;
 		Pdef.clear;
+	}
 
+	test_EventPatternProxy_xfade_cleanup_not_premature {
```

Comment:
this should go in a new file, `TestEventPatternProxy`

Reply:
TestPatternProxy.sc tests a whole bunch of JITlib classes including Pdef etc. that are even one layer above  EventPatternProxy [have a look](https://github.com/supercollider/supercollider/blob/4d1a4495b57232573d3c182220533ef13450b6f2/testsuite/classlibrary/TestPatternProxy.sc).

The file already include tests for EventPatternProxy, see [test_transparency_withEvents for example](https://github.com/supercollider/supercollider/blob/4d1a4495b57232573d3c182220533ef13450b6f2/testsuite/classlibrary/TestPatternProxy.sc#L67).

So I doubt a separate file is justified for this one test, unless you plan to break that TestPatternProxy.sc file into many.

Will do on the code style changes.

Reply:
yes, i understand, but please put it in a separate file. the reason we go with this kind of convention is because it is easier to decide on a simple, direct approach than to reason about every case separately. then there is no need to justify anything, really.

Reply:
and yes, there is a lot of existing technical debt in the test code, i.e. it isn't very consistent; that is why i'm trying to insist on doing things right in new code.

Reply:
Ok, separate file then. By the way, the main reason for the inner function is/was that when switching from a UnitTestScript (which is how I prototype the tests) to a class-based UnitTest one has to change every test expression from something

```
{ |test|

	var test_EventPatternProxy_xfade_cleanup_not_premature = {

		test.assert(....);
		test.assert(....);
	}
	;
	test_EventPatternProxy_xfade_cleanup_not_premature.();
}
```
to
```
SomeTest : UnitTest {
	
	test_EventPatternProxy_xfade_cleanup_not_premature {

		this.assert(....);
		this.assert(....);
	}
}
```

With that inner `assert` function defined up front (or at least some variable), there's only one place to replace `test` with `this` or vice-versa when converting between test class and test "script" (which is a function). Otherwise one needs to do a search-and-replace to convert back-and-forth.

The local function trick/technique was used in the rest of the TestPatternProxy file...

Reply:
And speaking of a separate file... I could move the `test_transparency_withEvents` method from `TestPatternProxy` to the new class (`TestEventPatternProxy`) too, since unlike `test_transparency` which tests `PatternProxy` in that regard, `test_transparency_withEvents` tests `EventPatternProxy`. 

The other two (later) methods presently found in `TestPatternProxy` (namely `test_update` and `test_update_withEvents`) alas test the updating via `Pdef`, `Pdefn` and even `Pbindef` rather than the layer-below JITlib classes that don't do dict lookups (meaning `PatternProxy` and `EventPatternProxy`, which also support updating though). So those two methods would be more difficult to relocate/refactor according to the guidelines, so I'm definitely punting on touching those in this PR. But, like I said, I could relocate `test_transparency_withEvents` since that's "just" an `EventPatternProxy`-test. For now, I'll only put my new method in the new class...

Reply:
> when switching from a UnitTestScript (which is how I prototype the tests) to a class-based UnitTest one has to change every test expression

this sounds like an issue in your workflow. there are other ways of doing this that require much less editing. for instance, you could use [this guard-sclang method](https://github.com/supercollider/supercollider/wiki/Unit-Testing-Guide#continuous-automatic-redgreen-testing-via-guard-sclang), or use `UnitTest.runTest`, or simply comment out the test methods you don't want to run.

> The local function trick/technique was used in the rest of the TestPatternProxy file...

i see that now. more technical debt :)

> For now, I'll only put my new method in the new class...

mhm, that's all you need to do here! where to put other tests, and how to work down the technical debt of the test suite in general, would be a great topic for a new issue. there's a lot in flight at the moment so i haven't had time to start that yet.

Reply:
Ok, I've pushed the changes